WordPress/Classes/ViewRelated/System/PagedViewController.swift

```diff
@@ -0,0 +1,178 @@
+            addChildViewController(controller)
+
+            controller.view.translatesAutoresizingMaskIntoConstraints = false
+            pagedStackView.addArrangedSubview(controller.view)
+
+            controller.view.shouldGroupAccessibilityChildren = true
+            controller.view.widthAnchor.constraintEqualToAnchor(view.widthAnchor, multiplier: 1.0).active = true
+            controller.view.topAnchor.constraintEqualToAnchor(view.topAnchor).active = true
+            controller.view.bottomAnchor.constraintEqualToAnchor(divider.topAnchor).active = true
+
+            controller.didMoveToParentViewController(self)
+        }
+    }
+
+    @IBAction func pageControlChanged() {
+        guard !scrollView.dragging else {
+            // If the user is currently dragging, reset the change and ignore it
+            pageControl.currentPage = currentIndex
+            return
+        }
+
+        // Stop the user interacting whilst we animate a scroll
+        scrollView.userInteractionEnabled = false
+
+        var targetPage = currentIndex
+        if pageControl.currentPage > currentIndex {
+            targetPage += 1
+        } else if pageControl.currentPage < currentIndex {
+            targetPage -= 1
+        }
+
+        scrollToPage(targetPage, animated: true)
+    }
+
+    private func updateForCurrentIndex() {
+        title = currentViewController.title
+
+        updatePageControl()
+
+        for (index, viewController) in viewControllers.enumerate() {
+            viewController.view.accessibilityElementsHidden = index != currentIndex
+        }
+    }
+
+    private func updatePageControl() {
+        pageControl?.currentPage = currentIndex
+    }
+
+    private var currentViewController: UIViewController {
+        return viewControllers[currentIndex]
+    }
+}
+
+extension PagedViewController: UIScrollViewDelegate {
+    func scrollViewDidScroll(scrollView: UIScrollView) {
+        // Ignore programmatic scrolling
+        if scrollView.dragging {
+            currentIndex = currentScrollViewPage()
+        }
+    }
+
+    override func accessibilityScroll(direction: UIAccessibilityScrollDirection) -> Bool {
+        var targetPage = currentIndex
+
+        switch direction {
+        case .Right: targetPage -= 1
+        case .Left: targetPage += 1
+        default: break
+        }
+
+        let success = scrollToPage(targetPage, animated: false)
+        if success {
+            accessibilityAnnounceCurrentPage()
+        }
+
+        return success
+    }
+
+    func scrollViewDidEndScrollingAnimation(scrollView: UIScrollView) {
+        scrollView.userInteractionEnabled = true
+
+        accessibilityAnnounceCurrentPage()
+    }
+
+    private func accessibilityAnnounceCurrentPage() {
+        guard let label = currentViewController.accessibilityLabel else {
+            return
+        }
+        UIAccessibilityPostNotification(UIAccessibilityLayoutChangedNotification, label)
```

Comment:
So it looks like we can post `currentViewController.view` to this method instead of the label to fix our accessibility problems :)